qiskit/optimization/problems/optimization_problem.py

```diff
@@ -342,227 +345,464 @@ def set_problem_type(self, _type):
-                                'Cannot substitute by variable that gets substituted itself.')
-                        w_ik = op.objective.get_quadratic_coefficients(i_name, k_name)
-                        w_ik += j_repl[1] * w_ij
-                        op.objective.set_quadratic_coefficients(i_name, k_name, w_ik)
-                elif i_repl is not None and j_repl is not None:
-                    if len(i_repl) == 1 and len(j_repl) == 1:
-                        offset += w_ij * i_repl[0] * j_repl[0] / 2
-                    elif len(i_repl) == 1 and len(j_repl) == 2:
-                        k = self.variables.get_indices(j_repl[0])
-                        k_name = self.variables.get_names(k)
-                        if k_name in vars_to_be_replaced.keys():
-                            raise QiskitOptimizationError(
-                                'Cannot substitute by variable that gets substituted itself.')
-                        w_k = op.objective.get_linear(k_name)
-                        w_k += w_ij * i_repl[0] * j_repl[1] / 2
-                        op.objective.set_linear(k_name, w_k)
-                    elif len(i_repl) == 2 and len(j_repl) == 1:
-                        k = self.variables.get_indices(i_repl[0])
-                        k_name = self.variables.get_names(k)
-                        if k_name in vars_to_be_replaced.keys():
-                            raise QiskitOptimizationError(
-                                'Cannot substitute by variable that gets substituted itself.')
-                        w_k = op.objective.get_linear(k_name)
-                        w_k += w_ij * j_repl[0] * i_repl[1] / 2
-                        op.objective.set_linear(k_name, w_k)
-                    else:  # both len(repl) == 2
-                        k = self.variables.get_indices(i_repl[0])
-                        k_name = self.variables.get_names(k)
-                        if k_name in vars_to_be_replaced.keys():
-                            raise QiskitOptimizationError(
-                                'Cannot substitute by variable that gets substituted itself.')
-                        m = self.variables.get_indices(j_repl[0])
-                        m_name = self.variables.get_names(m)
-                        if m_name in vars_to_be_replaced.keys():
-                            raise QiskitOptimizationError(
-                                'Cannot substitute by variable that gets substituted itself.')
-                        w_kl = op.objective.get_quadratic_coefficients(k_name, m_name)
-                        w_kl += w_ij * i_repl[1] * j_repl[1]
-                        op.objective.set_quadratic_coefficients(k_name, m_name, w_kl)
+                # substitute i <- j * v
+                # lb_i <= i <= ub_i  -->  lb_i / v <= j <= ub_i / v if v > 0
+                #                         ub_i / v <= j <= lb_i / v if v < 0
+                assert v != 0
```

Comment:
Could you add a `raise` statement with proper error message here?

Reply:
Done